cdap-ui/app/cdap/components/Cloud/Profiles/Store/ActionCreator.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import ProfilesStore, {PROFILES_ACTIONS} from 'components/Cloud/Profiles/Store';
+import {MyCloudApi} from 'api/cloud';
+import fileDownload from 'js-file-download';
+import {objectQuery} from 'services/helpers';
+
+export const getProfiles = (namespace) => {
+  ProfilesStore.dispatch({
+    type: PROFILES_ACTIONS.SET_LOADING,
+    payload: {
+      loading: true
+    }
+  });
+
+  MyCloudApi
+    .list({ namespace })
+    .subscribe(
+      (profiles) => {
+        ProfilesStore.dispatch({
+          type: PROFILES_ACTIONS.SET_PROFILES,
+          payload: { profiles }
+        });
+      },
+      (error) => {
+        ProfilesStore.dispatch({
+          type: PROFILES_ACTIONS.SET_ERROR,
+          payload: { error }
+        });
+      }
+    );
+};
+
+export const exportProfile = (namespace, profile) => {
+  MyCloudApi
+    .get({
+      namespace,
+      profile
+    })
+    .subscribe(
+      (res) => {
+        let json = JSON.stringify(res, null, 2);
+        let fileName = `${profile}-profile.json`;
+        fileDownload(json, fileName);
+      },
+      (error) => {
+        ProfilesStore.dispatch({
+          type: PROFILES_ACTIONS.SET_ERROR,
+          payload: { error }
+        });
+      }
+    );
+};
+
+export const deleteProfile = (namespace, profile) => {
+  let deleteObservable = MyCloudApi.delete({
+    namespace,
+    profile
+  });
+  deleteObservable.subscribe(() => {
+    getProfiles(namespace);
+  });
+  return deleteObservable;
+};
+
+export const importProfile = (namespace, e) => {
+  if (!objectQuery(e, 'target', 'files', 0)) {
+    return;
+  }
+
+  let uploadedFile = e.target.files[0];
+  let reader = new FileReader();
+  reader.readAsText(uploadedFile, 'UTF-8');
+
+  reader.onload =  (evt) => {
+    let jsonSpec = evt.target.result;
+    try {
+      jsonSpec = JSON.parse(jsonSpec);
+    } catch (error) {
+      ProfilesStore.dispatch({
+        type: PROFILES_ACTIONS.SET_ERROR,
+        payload: {
+          error: error.message || error
```

Comment:
I think this is `error.response` vs `error.message` I believe. Right now this is what the error I am seeing.

![screen shot 2018-04-23 at 11 06 29 am](https://user-images.githubusercontent.com/1452845/39144728-86f04518-46e6-11e8-99ec-45b900b12a1a.png)

Reply:
The error in the line you highlighted is actually for parsing JSON, and if there's an error then it is under `error.message`. The error that shows in the screenshot is actually logged a few lines below, in the error handler for `MyCloudApi.create`. I have changed that one to `error.response`